cishouseholds/edit.py

```diff
@@ -792,3 +793,29 @@ def edit_to_sum_or_max_value(
         .cast("integer"),
     )
     return df
+
+
+def add_work_from_home_identifier(
+    df: DataFrame,
+    columns_to_check_in: List[str],
+    regex_pattern: str = "(W(K|ORK.*?) F(ROM?) H(OME?))|(WFH)",
+    pattern_result_column: str = "is_wfh",
+):
+    """
+    Applies the user provided RegEx pattern to the list of columns. If a value in any of the
+    columns matches the RegEx pattern then `pattern_result_column` column will have the corresponding
+    value set to integer: 1 (True), 0 (False) otherwise.
+
+    Parameters:
+    -----------
+    df: The input dataframe to process
+    columns_to_check_in: a list of columns in which to look for the `regex_pattern`
+    regex_pattern: the Spark-compatible regex pattern to check against
+    pattern_result_column: name of the output column which will contain the result of the RegEx pattern search
+
+    """
+    df = df.withColumn(
+        pattern_result_column, any_column_matches_regex(columns_to_check_in, regex_pattern).cast("integer")
```

Comment:
Thinking about how we're gonna use this column, it might be worth keeping it as a Boolean? I'm guessing that we'll create a series of these columns and then have some logic like: "when working_from_home or unemployed", which will read nicely with Boolean cols